commerce-service/src/main/java/com/liferay/commerce/service/impl/CommerceOrderItemServiceImpl.java

```diff
@@ -288,6 +293,35 @@ public CommerceOrderItem updateCommerceOrderItemInfo(
 			requestedDeliveryDateMinute, serviceContext);
 	}
 
+	@Override
+	public CommerceOrderItem updateCommerceOrderItemPrices(
+			long commerceOrderItemId, BigDecimal unitPrice,
+			BigDecimal promoPrice, BigDecimal discountAmount,
+			BigDecimal finalPrice, BigDecimal discountPercentageLevel1,
+			BigDecimal discountPercentageLevel2,
+			BigDecimal discountPercentageLevel3,
+			BigDecimal discountPercentageLevel4)
+		throws PortalException {
+
+		CommerceOrderItem commerceOrderItem =
+			commerceOrderItemLocalService.getCommerceOrderItem(
+				commerceOrderItemId);
+
+		CommerceOrder commerceOrder = commerceOrderItem.getCommerceOrder();
```

Comment:
@alecsloan We should just check if the user can update the order using the order permission, that already checks for guest visibility

Reply:
In the upsert method commercePrdocutViewPermission still has to be checked to see if they can view the product?